src/Microsoft.AspNetCore.Mvc.Core/Formatters/MediaType.cs

```diff
@@ -292,6 +292,12 @@ public static Encoding GetEncoding(StringSegment mediaType)
         public static MediaTypeSegmentWithQuality CreateMediaTypeSegmentWithQuality(string mediaType, int start)
         {
             var parsedMediaType = new MediaType(mediaType, start, length: null);
+            if (parsedMediaType.Type.Equals(default(StringSegment)) ||
```

Comment:
Would it help to change the ctor to use `default(StringSegment)` - https://github.com/aspnet/Mvc/blob/6be4eb1fd654b1c1ea281416c85d407846c9ae2d/src/Microsoft.AspNetCore.Mvc.Core/Formatters/MediaType.cs#L69? This should work as is, but it might make it easier to reason about.